### PR TITLE
Find the app bundle when the flavor contains underscores

### PR DESCRIPTION
**Pull request by blasten**
## Description

The tool can't find app bundles when the `--flavor` flag contains underscores.

## Related Issues

https://github.com/flutter/flutter/issues/36067

## Tests

I added the following tests:

* gradle_plugin_bundle_test.dart

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/gradle.dart [hunk 1]
```diff
@@ -822,19 +822,22 @@ Iterable<File> findApkFiles(GradleProject project, AndroidBuildInfo androidBuild
...
-  File bundleFile = project.bundleDirectory.childDirectory(modeName).childFile(bundleFileName);
-  if (bundleFile.existsSync())
+  }
+  File bundleFile = project.bundleDirectory
+    .childDirectory(camelCase(buildInfo.modeName))
+    .childFile(bundleFileName);
+  if (bundleFile.existsSync()) {
     return bundleFile;
+  }
   if (buildInfo.flavor != null) {
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/android/gradle.dart]: nit:
```dart
if (buildInfo.flavor == null) {
  return null;
}
```
then un-indent.

File: packages/flutter_tools/lib/src/android/gradle.dart [hunk 2]
```diff
@@ -822,19 +822,22 @@ Iterable<File> findApkFiles(GradleProject project, AndroidBuildInfo androidBuild
 
 File _findBundleFile(GradleProject project, BuildInfo buildInfo) {
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/android/gradle.dart]: Can you make this `@visibleForTesting` and add unit tests? I see gradle_test.dart already has tests for e.g. `findApkFiles` above.

File: packages/flutter_tools/lib/src/android/gradle.dart (hunk 1 above)
Comment (blasten) [reply packages/flutter_tools/lib/src/android/gradle.dart]: Done.

File: packages/flutter_tools/lib/src/android/gradle.dart (hunk 2 above)
Comment (blasten) [reply packages/flutter_tools/lib/src/android/gradle.dart]: I added a few unit tests :)

**Review by zanderso — APPROVED** [review]

**Pull request merged**